src/base/dev/vga/vgaemu.c

```diff
@@ -805,15 +803,19 @@ unsigned vga_read_dword(dosaddr_t addr)
 void vga_mark_dirty(dosaddr_t s_addr, int len)
 {
   unsigned vga_page, abeg, aend, addr;
+  int i, j;
   abeg = s_addr & PAGE_MASK;
   aend = (s_addr + len - 1) & PAGE_MASK;
   for (addr = abeg; addr <= aend; addr += PAGE_SIZE) {
-    if (!vga_write_access(addr))
-      continue;
-    vga_page = (addr >> PAGE_SHIFT) -
-	vga.mem.map[VGAEMU_MAP_BANK_MODE].base_page +
-	vga.mem.map[VGAEMU_MAP_BANK_MODE].first_page;
-    _vga_emu_adjust_protection(vga_page, 0, VGA_PROT_RW, 1);
+    for(i = 0; i < VGAEMU_MAX_MAPPINGS; i++) {
+      j = (addr >> PAGE_SHIFT) - vga.mem.map[i].base_page;
+      if(j >= 0 && j < vga.mem.map[i].pages) {
+	vga_page = j + vga.mem.map[i].first_page;
+	break;
```

Comment:
What's wring with indentation here?

Reply:
If indent is 2 spaces but you use tab=8 spaces that's how github renders it, I can just replace the tab with 8 spaces.